rest/handler.go

```diff
@@ -283,27 +283,36 @@ func (h *handler) logStatus(status int, message string) {
 	h.logDuration(false) // don't track actual time
 }
 
-func (h *handler) checkAuth(context *db.DatabaseContext) error {
+func (h *handler) checkAuth(context *db.DatabaseContext) (err error) {
+
 	h.user = nil
 	if context == nil {
 		return nil
 	}
 
 	// Record TotalAuthTime stat
 	defer func(t time.Time) {
-		if context.DbStats == nil {
-			return
-		}
 		delta := time.Since(t).Nanoseconds()
 		context.DbStats.StatsSecurity().Add(base.StatKeyTotalAuthTime, delta)
 	}(time.Now())
 
-	var err error
+	defer func() {
+		if h.db == nil || h.db.DatabaseContext == nil ||  h.db.DatabaseContext.DbStats == nil {
+			return
+		}
+		if err != nil {
+			h.db.DatabaseContext.DbStats.StatsSecurity().Add(base.StatKeyAuthFailedCount, 1)
+		} else {
+			h.db.DatabaseContext.DbStats.StatsSecurity().Add(base.StatKeyAuthSuccessCount, 1)
+		}
+	}()
```

Comment:
Seems cleaner to just have a single defer for stats processing - looks like this could move up to the previous defer, and use context.DbStats instead of h.db.DatabaseContext.DbStats to avoid the nil checks.

Reply:
Fixed